libraries/chain/controller.cpp

```diff
@@ -2854,16 +2854,17 @@ struct controller_impl {
 
    static std::optional<qc_data_t> extract_qc_data(const signed_block_ptr& b) {
```

Comment:
Seems we should name this `validate_and_extract_qc_data` as it is calling the `validate_and_extract` methods of extensions.

Reply:
Hum, I think it could be misleading, it does the generic extension validation, but not the actual `qc_data` validation implemented in `verify_qc_claim`.